plugins/AtlasExplorer/api/usecases/FlatmapAreas.cpp

```diff
@@ -0,0 +1,175 @@
+            areaList.push_back(index);
+        }
+
+        return map;
+    }
+};
+
+class PrimitiveBuilder
+{
+public:
+    static std::vector<std::vector<brayns::Box>> build(const FlatmapAtlas &atlas)
+    {
+        auto map = AreaMapGenerator::fromAtlas(atlas);
+
+        auto geometry = std::vector<std::vector<brayns::Box>>();
+        geometry.reserve(_countTotalAreas(map));
+
+        for (auto &[x, subMap] : map)
+        {
+            for (auto &[y, areaVoxelIndices] : subMap)
+            {
+                auto areaVoxels = _generatePrimitives(atlas, areaVoxelIndices);
+                geometry.push_back(std::move(areaVoxels));
+            }
+        }
+        return geometry;
+    }
+
+private:
+    static size_t _countTotalAreas(const AreaMap &map)
+    {
+        size_t result = 0;
+
+        for (auto &[x, subMap] : map)
+        {
+            result += subMap.size();
+        }
+        return result;
+    }
+
+    static std::vector<brayns::Box> _generatePrimitives(const FlatmapAtlas &atlas, const std::vector<size_t> &indices)
+    {
+        auto result = std::vector<brayns::Box>();
+        result.reserve(indices.size());
+
+        for (auto index : indices)
+        {
+            auto bounds = atlas.getVoxelBounds(index);
+            result.push_back(brayns::Box{bounds.getMin(), bounds.getMax()});
+        }
+        return result;
+    }
+};
+
+class ModelBuilder
+{
+public:
+    static std::unique_ptr<brayns::Model> build(std::vector<std::vector<brayns::Box>> primitives)
+    {
+        auto model = std::make_unique<brayns::Model>();
+
+        auto &components = model->getComponents();
+
+        auto &geometries = components.add<brayns::Geometries>();
+        geometries.elements.reserve(primitives.size());
+        auto &views = components.add<brayns::GeometryViews>();
+        views.elements.reserve(primitives.size());
+        auto roulette = brayns::ColorRoulette();
+        for (auto &primitiveList : primitives)
+        {
+            auto &geometry = geometries.elements.emplace_back(std::move(primitiveList));
+            auto &view = views.elements.emplace_back(geometry);
+            view.setColor(roulette.getNextColor());
+        }
+
+        auto &systems = model->getSystems();
+        systems.setBoundsSystem<brayns::GenericBoundsSystem<brayns::Geometries>>();
+        systems.setInitSystem<brayns::GeometryInitSystem>();
+        systems.setCommitSystem<brayns::GeometryCommitSystem>();
+
+        return model;
+    }
+};
+}
+
+std::string FlatmapAreas::getName() const
+{
+    return "Flatmap areas";
+}
+
+bool FlatmapAreas::isAtlasValid(const Atlas &atlas) const
+{
+    return atlas.getVoxelType() == VoxelType::flatmap;
+}
+
+std::unique_ptr<brayns::Model> FlatmapAreas::run(const Atlas &atlas, const brayns::JsonValue &payload) const
+{
+    (void)payload;
+    assert(dynamic_cast<const FlatmapAtlas *>(&atlas));
+
+    auto &flatmapAtlas = static_cast<const FlatmapAtlas &>(atlas);
```

Comment:
Maybe at some point we can make a utility function like cast<T>(value) {assert(dynamic), return static}.